src/raft.c

```diff
@@ -1632,9 +1715,33 @@ void RaftReqFree(RaftReq *req)
         RedisModule_FreeThreadSafeContext(req->ctx);
         RedisModule_UnblockClient(req->client, NULL);
     }
+
     RedisModule_Free(req);
 }
 
+static void blockedTimedOut(RedisModuleCtx *ctx, void *data)
+{
+    RedisRaftCtx *rr = &redis_raft;
+    RaftReq *req = (RaftReq *) data;
+
+    raft_entry_t *entry = raft_entry_new(sizeof(raft_index_t));
+    entry->id = rand();
+    entry->type = RAFT_LOGTYPE_TIMEOUT_BLOCKED;
+    memcpy(entry->data, &req->raft_idx, entry->data_len);
+    entryAttachRaftReq(rr, entry, req);
+
+    int e = raft_recv_entry(rr->raft, entry, NULL);
+    if (req && e != 0) {
+        /* TODO: unsure if we should reply to user here, like in other case.
+         * I'd think in this case, it would be the timer went off and we aren't leader anymore,
+         * which will be more cleanly handled upon new term
+         */
```

Comment:
It might be better to reply here. If we can't elect a new leader, client will stay blocked forever.
Still, we can append successfully and lose leadership before replicating/applying the entry. Then, client will stay blocked.

Not sure if this is important but maybe something to think about. Wdyt?

Reply:
so I think its actually wrong to reply here, and updated the commen.

If we can't append, three things could happen
 
this entry has been comitted, so is pending at timeout time.
1) has in practice executed before term change, therefore doesn't matter
2) hasn't executed at term change, will be aborted by term change
 
this entry hasn't been comitted
 
3) entry will be popped and aborted via pop of log.

Reply:
No, my point was if this node won't see a new term (cluster is down, this node becomes isolated etc), it will never pop the entry and it won't unblock client. 

I guess clients should be prepared for these anyway. Probably either user will set a timeout or client will apply it is own operation timeout on its local.

Reply:
my point is that we shouldn't have 2 code paths that WILL be executed that respond to the client.  While we could respond here, it means we have to code to prevent the other 2 possibilities from happening which is ugly.